module/pollSubmissions/helper.js

```diff
@@ -0,0 +1,315 @@
+                                incrementKeyValues[`result.${singleResponse.qid}.${value}`] = 1;
+                            })
+                        }
+                        else {
+                            incrementKeyValues[`result.${singleResponse.qid}.${singleResponse.value}`] = 1;
+                        }
+                    })
+
+                    let updateQuery = { 
+                        $inc: incrementKeyValues
+                    }
+
+                    await pollsHelper.updatePollDocument(pollId, updateQuery)
+            
+                    return resolve({
+                        success: true,
+                        message: messageConstants.apiResponses.POLL_SUBMITTED,
+                        data: true
+                    });
+
+            } catch (error) {
+                return resolve({
+                    success: false,
+                    message: error.message,
+                    data: false
+                });
+            }
+        });
+    }
+
+
+    /**
+    * List all the polls.
+    * @method
+    * @name list
+    * @param {String} userId - userId
+    * @returns {JSON} - Polls list.
+    */
+
+    static list(userId = "") {
+        return new Promise(async (resolve, reject) => {
+            try {
+
+                if (userId == "") {
+                    throw new Error(messageConstants.apiResponses.USER_ID_REQUIRED_CHECK);
+                }
+
+                let getPollList = [
+                    pollsHelper.list
+                        (
+                            userId
+                        ),
+                    database.models.pollSubmissions.aggregate
+                        (
+                            [
+                                { "$match": { "userId": userId, isDeleted: false } },
+                                {
+                                    "$project": {
+                                        '_id': "$pollId",
+                                        'name': '$pollName'
+                                    }
+                                },
+                                { "$sort": { createdAt: 1 } }
+                            ])
+                ];
+
+
+                let result = [];
+
+                await Promise.all(getPollList)
+                    .then(function (response) {
+
+                        if (Array.isArray(response[0].data) && response[0].data.length > 0) {
+                            result = result.concat(response[0].data);
+                        }
+
+                        if (response[1].length > 0) {
+                            result = result.concat(response[1]);
+                        }
+                    });
+
+                let uniquePolls = [];
+
+                if (result.length > 0) {
+
+                    let mymap = new Map();
+                    uniquePolls = result.filter(el => {
+                        const val = mymap.get(el.name);
```

Comment:
@deepakr123 - We should not use the name as unique, go with id, what I shared was just for reference. Better would be to create gen.utils function which anybody can be used later on. The function takes an array as an argument and the key which should be unique in an array of objects.

Reply:
resolved